### PR TITLE
bpo-37120: Add SSLContext.num_tickets (GH-13719)

### PR DESCRIPTION
**Pull request by tiran**
Signed-off-by: Christian Heimes <christian@python.org>

https://bugs.python.org/issue37120

### CODE REVIEW

File: Doc/library/ssl.rst [hunk 1]
```diff
@@ -1959,6 +1959,18 @@ to speed up repeated connections from the same clients.
...
+
+   Control the number of TLSv1.3 session tickets of a
+   :attr:`TLS_PROTOCOL_SERVER` context.
+
+   .. note::
+
+     This attribute is not available unless the ssl module is compiled
+     with OpenSSL 1.1.1 or newer.
+
+   .. versionadded:: 3.8
```
Comment (njsmith) [inline Doc/library/ssl.rst:1973]: Probably worth noting explicitly that it only affects TLS 1.3 connections, and possibly that it doesn't affect connections that are resumed using a ticket.

File: Lib/test/test_ssl.py [hunk 2]
```diff
@@ -1634,6 +1634,24 @@ class MySSLObject(ssl.SSLObject):
...
+        self.assertEqual(ctx.num_tickets, 1)
+        ctx.num_tickets = 0
+        self.assertEqual(ctx.num_tickets, 0)
+        with self.assertRaises(ValueError):
+            ctx.num_tickets = -1
+        with self.assertRaises(TypeError):
+            ctx.num_tickets = None
+
+        ctx = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+        self.assertEqual(ctx.num_tickets, 2)
```
Comment (njsmith) [inline Lib/test/test_ssl.py:1651]: It's maybe slightly nicer to make this error out too, I guess?

**Review by njsmith — COMMENTED** [review]
Two small comments.

File: Doc/library/ssl.rst (hunk 1 above)
Comment (tiran) [reply Doc/library/ssl.rst:1973]: SGTM, the first sentence mentions that it's about TLS 1.3. I added another sentence to explicitly state that the setting does not affect TLS 1.0 to 1.2.

File: Lib/test/test_ssl.py (hunk 2 above)
Comment (tiran) [reply Lib/test/test_ssl.py:1651]: You mean getattr access? I'm not sure how much trouble it's going to create.

**Pull request merged**